### PR TITLE
Add ambari plugin #1469

### PR DESCRIPTION
**Pull request by githubzhangchunyang**
Add ambari plugin #1469

### CODE REVIEW

File: ambari_plugin/common-services/DOLPHIN/1.2.1/package/scripts/status_params.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Redis service params
+
```
Comment (Jave-Chen) [inline ambari_plugin/common-services/DOLPHIN/1.2.1/package/scripts/status_params.py]: license description needed

**Comment by dailidong** [comment]
amazing

File: ambari_plugin/common-services/DOLPHIN/1.2.1/package/scripts/dolphin_alert_service.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        env.set_params(params)
+
+        dolphin_env()
+
+    def start(self, env):
+        import params
+        env.set_params(params)
+        self.configure(env)
+        start_cmd = format("sh " + params.dolphin_bin_dir + "/dolphinscheduler-daemon.sh start alert-server")
+        Execute(start_cmd, user=params.dolphin_user)
```
Comment (gaojun2048) [inline ambari_plugin/common-services/DOLPHIN/1.2.1/package/scripts/dolphin_alert_service.py]: Have you tested what happens if the service is already started at this time and then click Start? Do you need to add the not_if parameter.

File: ambari_plugin/common-services/DOLPHIN/1.2.1/package/scripts/dolphin_worker_service.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+
```
Comment (gaojun2048) [inline ambari_plugin/common-services/DOLPHIN/1.2.1/package/scripts/dolphin_worker_service.py]: License is need

File: ambari_plugin/common-services/DOLPHIN/1.2.1/package/scripts/dolphin_alert_service.py (hunk 2 above)
Comment (githubzhangchunyang) [reply ambari_plugin/common-services/DOLPHIN/1.2.1/package/scripts/dolphin_alert_service.py]: OK! I will

File: ambari_plugin/common-services/DOLPHIN/1.2.1/metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            <name>DOLPHIN</name>
+            <displayName>Dolphin Scheduler</displayName>
+            <comment>分布式易扩展的可视化DAG工作流任务调度系统</comment>
+            <version>1.2.1</version>
+            <components>
+                <component>
+                    <name>DOLPHIN_MASTER</name>
+                    <displayName>DS Master</displayName>
+                    <category>MASTER</category>
+                    <cardinality>1-2</cardinality>
```
[the remark this answers is not in the document]
Comment (githubzhangchunyang) [reply ambari_plugin/common-services/DOLPHIN/1.2.1/metainfo.xml]: yeah!  The master component supports multiple installations

**Comment by dailidong** [comment]
@Jave-Chen ，can recheck again?

**Comment by gaojun2048** [comment]
+1

**Comment by barntt77** [comment]
I want to make the source code into RPM package，but have a question。
image
thank for your help！

**Comment by barntt77** [comment]
> I want to make the source code into RPM package，but have a question。
> image
> thank for your help！

[INFO] dolphinscheduler ................................... SUCCESS [  4.839 s]
[INFO] dolphinscheduler-ui ................................ SUCCESS [  1.730 s]
[INFO] dolphinscheduler-common ............................ SUCCESS [ 31.917 s]
[INFO] dolphinscheduler-dao ............................... SUCCESS [ 29.429 s]
[INFO] dolphinscheduler-rpc ............................... SUCCESS [ 17.768 s]
[INFO] dolphinscheduler-alert ............................. SUCCESS [ 14.196 s]
[INFO] dolphinscheduler-server ............................ SUCCESS [ 23.985 s]
[INFO] dolphinscheduler-api ............................... SUCCESS [ 31.803 s]
[INFO] dolphinscheduler-dist .............................. FAILURE [ 21.314 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:03 min
[INFO] Finished at: 2020-02-27T16:24:10+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:rpm-maven-plugin:2.2.0:attached-rpm (default) on project dolphinscheduler-dist: Source location C:\Users\ZhiFa\Desktop\idea\apache-dolphinscheduler-incubating-1.2.1-src-release\dolphinscheduler-dist\..\dolphinschedul
er-ui\dist does not exist -> [Help 1]

**Comment by gaojun2048** [comment]
> > I want to make the source code into RPM package，but have a question。
> > image
> > thank for your help！
> 
> [INFO] dolphinscheduler ................................... SUCCESS [ 4.839 s]
> [INFO] dolphinscheduler-ui ................................ SUCCESS [ 1.730 s]
> [INFO] dolphinscheduler-common ............................ SUCCESS [ 31.917 s]
> [INFO] dolphinscheduler-dao ............................... SUCCESS [ 29.429 s]
> [INFO] dolphinscheduler-rpc ............................... SUCCESS [ 17.768 s]
> [INFO] dolphinscheduler-alert ............................. SUCCESS [ 14.196 s]
> [INFO] dolphinscheduler-server ............................ SUCCESS [ 23.985 s]
> [INFO] dolphinscheduler-api ............................... SUCCESS [ 31.803 s]
> [INFO] dolphinscheduler-dist .............................. FAILURE [ 21.314 s]
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD FAILURE
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time: 03:03 min
> [INFO] Finished at: 2020-02-27T16:24:10+08:00
> [INFO] ------------------------------------------------------------------------
> [ERROR] Failed to execute goal org.codehaus.mojo:rpm-maven-plugin:2.2.0:attached-rpm (default) on project dolphinscheduler-dist: Source location C:\Users\ZhiFa\Desktop\idea\apache-dolphinscheduler-incubating-1.2.1-src-release\dolphinscheduler-dist..\dolphinschedul
> er-ui\dist does not exist -> [Help 1]


Hi, From the log it looks like the dir `C:\Users\ZhiFa\Desktop\idea\apache-dolphinscheduler-incubating-1.2.1-src-release\dolphinscheduler-dist..\dolphinscheduler-ui\dist` is not exist.
Can you take a look at this directory？

Under normal circumstances, you need to install the rpm-build tool on the environment before running the mvn command to play the rpm package. 

To install rpm-build tools in centos , you can do like this:

```
yum install –y rpm-build rpmdevtools.noarch
echo “%_topdir $YOURDIR” >~/.rpmmacros
rpmdev-setuptree
```

**Comment by barntt77** [comment]
> > > I want to make the source code into RPM package，but have a question。
> > > image
> > > thank for your help！
> > 
> > 
> > [INFO] dolphinscheduler ................................... SUCCESS [ 4.839 s]
> > [INFO] dolphinscheduler-ui ................................ SUCCESS [ 1.730 s]
> > [INFO] dolphinscheduler-common ............................ SUCCESS [ 31.917 s]
> > [INFO] dolphinscheduler-dao ............................... SUCCESS [ 29.429 s]
> > [INFO] dolphinscheduler-rpc ............................... SUCCESS [ 17.768 s]
> > [INFO] dolphinscheduler-alert ............................. SUCCESS [ 14.196 s]
> > [INFO] dolphinscheduler-server ............................ SUCCESS [ 23.985 s]
> > [INFO] dolphinscheduler-api ............................... SUCCESS [ 31.803 s]
> > [INFO] dolphinscheduler-dist .............................. FAILURE [ 21.314 s]
> > [INFO] ------------------------------------------------------------------------
> > [INFO] BUILD FAILURE
> > [INFO] ------------------------------------------------------------------------
> > [INFO] Total time: 03:03 min
> > [INFO] Finished at: 2020-02-27T16:24:10+08:00
> > [INFO] ------------------------------------------------------------------------
> > [ERROR] Failed to execute goal org.codehaus.mojo:rpm-maven-plugin:2.2.0:attached-rpm (default) on project dolphinscheduler-dist: Source location C:\Users\ZhiFa\Desktop\idea\apache-dolphinscheduler-incubating-1.2.1-src-release\dolphinscheduler-dist..\dolphinschedul
> > er-ui\dist does not exist -> [Help 1]
> 
> Hi, I test the rpm command and found it`s a bug. Can you create a issue about it ? I will fix this bug.

I'll create a issue about it immediately

**Comment by barntt77** [comment]
> > > > I want to make the source code into RPM package，but have a question。
> > > > image
> > > > thank for your help！
> > > 
> > > 
> > > [INFO] dolphinscheduler ................................... SUCCESS [ 4.839 s]
> > > [INFO] dolphinscheduler-ui ................................ SUCCESS [ 1.730 s]
> > > [INFO] dolphinscheduler-common ............................ SUCCESS [ 31.917 s]
> > > [INFO] dolphinscheduler-dao ............................... SUCCESS [ 29.429 s]
> > > [INFO] dolphinscheduler-rpc ............................... SUCCESS [ 17.768 s]
> > > [INFO] dolphinscheduler-alert ............................. SUCCESS [ 14.196 s]
> > > [INFO] dolphinscheduler-server ............................ SUCCESS [ 23.985 s]
> > > [INFO] dolphinscheduler-api ............................... SUCCESS [ 31.803 s]
> > > [INFO] dolphinscheduler-dist .............................. FAILURE [ 21.314 s]
> > > [INFO] ------------------------------------------------------------------------
> > > [INFO] BUILD FAILURE
> > > [INFO] ------------------------------------------------------------------------
> > > [INFO] Total time: 03:03 min
> > > [INFO] Finished at: 2020-02-27T16:24:10+08:00
> > > [INFO] ------------------------------------------------------------------------
> > > [ERROR] Failed to execute goal org.codehaus.mojo:rpm-maven-plugin:2.2.0:attached-rpm (default) on project dolphinscheduler-dist: Source location C:\Users\ZhiFa\Desktop\idea\apache-dolphinscheduler-incubating-1.2.1-src-release\dolphinscheduler-dist..\dolphinschedul
> > > er-ui\dist does not exist -> [Help 1]
> > 
> > 
> > Hi, I test the rpm command and found it`s a bug. Can you create a issue about it ? I will fix this bug.
> 
> I'll create a issue about it immediately

issues  #2038

**Pull request merged**